[fix] [broker] correct compaction phase one timeout meaning

thetumbled opened the pull request:
### Motivation
Configuratioin `brokerServiceCompactionPhaseOneLoopTimeInSeconds` is defined as the total timeout of phase one.
```
    @FieldContext(
            category = CATEGORY_SERVER,
            doc = "Timeout for the compaction phase one loop, If the execution time of the compaction "
                    + "phase one loop exceeds this time, the compaction will not proceed."
    )
    private long brokerServiceCompactionPhaseOneLoopTimeInSeconds = 30;
```
But current implementation just set a timeout for a single message read operation, which is different with the meaning of `brokerServiceCompactionPhaseOneLoopTimeInSeconds` and will result into thousands of timer tasks be scheduled and cancelled in the scheduler because we may iterate many messages.

### Modifications
Set a timer task for the tatol phase one.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.


### Does this pull request potentially affect one of the following parts:

<!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

<!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->

- [ ] `doc` <!-- Your PR contains doc changes. -->
- [ ] `doc-required` <!-- Your PR changes impact docs and you will update later -->
- [x] `doc-not-needed` <!-- Your PR changes do not impact docs -->
- [ ] `doc-complete` <!-- Docs have been already added -->

### Matching PR in forked repository

PR in forked repository: https://github.com/thetumbled/pulsar/pull/32

poorbarcode commented on pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java:
```diff
@@ -120,9 +123,6 @@ private void phaseOneLoop(RawReader reader,
             return;
         }
         CompletableFuture<RawMessage> future = reader.readNextAsync();
-        FutureUtil.addTimeoutHandling(future,
```
The scheduled task will be canceled when the `future` is complete, and the second scheduled task will only be created after the first `future` is completed, so it will not cause thousands of scheduled tasks.

thetumbled replied:
Originally i am worried that the frequent tasks schedule and cancellation will impact the performance, but it seems not.

poorbarcode commented on pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java:
```diff
@@ -91,6 +91,9 @@ protected CompletableFuture<Long> doCompaction(RawReader reader, BookKeeper bk)
     private CompletableFuture<PhaseOneResult> phaseOne(RawReader reader) {
         Map<String, MessageId> latestForKey = new HashMap<>();
         CompletableFuture<PhaseOneResult> loopPromise = new CompletableFuture<>();
+        FutureUtil.addTimeoutHandling(loopPromise,
```
maybe `30s` is not enough for the step one

thetumbled replied:
It does, it is predictable for the time of reading one single message, but not for executing whole phase one.

codelipenghui commented:
I think I have introduced a bad configuration name in https://github.com/apache/pulsar/pull/11206. Actually, it was the timeout for an entry read. 

IMO, we should have a new configuration name and deprecate the existing one. I don't think it's reasonable to have a timeout for the whole phase one loop. After the first time the phase one loop get timeout, the subsequent phase one loop will almost always get timeout with more published messages unless changing phase one loop timeout.

thetumbled commented:
> I think I have introduced a bad configuration name in #11206. Actually, it was the timeout for an entry read.
> 
> IMO, we should have a new configuration name and deprecate the existing one. I don't think it's reasonable to have a timeout for the whole phase one loop. After the first time the phase one loop get timeout, the subsequent phase one loop will almost always get timeout with more published messages unless changing phase one loop timeout.

I agree. I have introduced a new configuration name and deprecate the existing one. 
PTAL, thanks.

thetumbled commented:
I have created a pip to change the conf name, please help review it, thanks! 
https://github.com/apache/pulsar/pull/21910
@codelipenghui @poorbarcode @BewareMyPower

thetumbled commented:
I have refactor the implementation to just modify the doc, so that we can avoid to create a pip and introduce a new conf name. PTAL, thanks.
@codelipenghui @poorbarcode @BewareMyPower

poorbarcode commented:
/pulsarbot rerun-failure-checks

thetumbled commented:
/pulsarbot rerun-failure-checks
